src/Toml/Parser.hs

```diff
@@ -79,12 +80,66 @@ basicStringP :: Parser Text
 basicStringP = lexeme $ Text.pack <$> (char '"' *> anyChar `manyTill` char '"')
 
 textP :: Parser Text
-textP = literalStringP <|> basicStringP
+textP = multilineStringP <|> multilineLiteralStringP <|> literalStringP <|> basicStringP
 
 -- adds " or ' to both sides
 quote :: Text -> Text -> Text
 quote q t = q <> t <> q
 
+nonControlChar :: Parser Text
+nonControlChar = Text.singleton <$> satisfy (not . isControl)
+
+multilineStringP :: Parser Text
+multilineStringP = lexeme $ fmap mconcat $ quotes3
+                                        >> optional eol
+                                        >> allowedChar `manyTill` quotes3
+  where
+    quotes3 = string "\"\"\""
+
+    allowedChar :: Parser Text
+    allowedChar = lineEndingBackslash <|> escapeSequence <|> nonControlChar <|> eol
+
+    lineEndingBackslash :: Parser Text
+    lineEndingBackslash = Text.empty <$ try (char '\\' >> eol >> space)
+
+    escapeSequence :: Parser Text
+    escapeSequence = char '\\' >> anyChar
+                 >>= \case
+                      'b'  -> return "\b"
+                      't'  -> return "\t"
```

Comment:
It's better to use `pure` instead of `return`, because it's shorter 😏

Reply:
That's a good reason to use `pure`, that I haven't thought of before.